ios/FluentUI/Core/Theme/Tokens/ColorSet.swift

```diff
@@ -5,10 +5,49 @@
 
 import SwiftUI
 
-/// An alias that represents a three-channel, 8-bit-per-channel color value, usually in hex.
-///
-/// For example: `0xFF0000` represents red, `0x00FF00` green, and `0x0000FF` blue.
-public typealias ColorValue = UInt32
+/// A platform-agnostic representation of a 32-bit RGBA color value.
+public struct ColorValue {
+    var r: CGFloat { CGFloat((hexValue & 0xFF000000) >> 24) / 255.0 }
+    var g: CGFloat { CGFloat((hexValue & 0x00FF0000) >> 16) / 255.0 }
+    var b: CGFloat { CGFloat((hexValue & 0x0000FF00) >> 8) / 255.0 }
+    var a: CGFloat { CGFloat(hexValue & 0x000000FF) / 255.0 }
+
+    /// Creates a color value instance with the specified three-channel, 8-bit-per-channel color value, usually in hex.
+    ///
+    /// For example: `0xFF0000` represents red, `0x00FF00` green, and `0x0000FF` blue.
+    /// There is no way to specify an alpha channel via this initializer. For that, use `init(r:g:b:a)` instead.
```

Comment:
Does this silently accepts an unintended different color if the alpha is specified?
Is there a way we can add a validation that this fails if the value added specifies an alpha value?

Reply:
Hmm, not sure what you mean. Can you provide an example of a value you'd be concerned of?

My thought is that if you specify as ARGB, the A gets bit-shifted off the end and ignored. And if you specify as RGBA, well then yeah it'll break, but that can't really be helped.

Reply:
My comment was based on the assumption of the RGBA (probably with the order of the parameters in the back of my mind).
I agree with your thought below:
`My thought is that if you specify as ARGB, the A gets bit-shifted off the end and ignored.`